Tests/Common/Data/DividendYieldProviderTests.cs

```diff
@@ -0,0 +1,42 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using NUnit.Framework;
+using QuantConnect.Data;
+
+namespace QuantConnect.Tests.Common.Data
+{
+    [TestFixture]
+    public class DividendYieldProviderTests
+    {
+        [TestCase("19700306", 0.0)]   // Date in before the first date in file
+        [TestCase("20200205", 0.04147)]
+        [TestCase("20200206", 0.03355)]
+        [TestCase("20200207", 0.03355)]
+        [TestCase("20210203", 0.01676)]
+        [TestCase("20210204", 0.01239)]
+        [TestCase("20210205", 0.01239)]
+        [TestCase("20491231", 0.01239)]   // Date in far future, assuming same rate
+        public void GetDividendYieldRate(string dateString, double expected)
```

Comment:
Nice 👍  let's add a unit test for a case for which we have no factor file data to assert it's handled happily